src/Parsers/ExpressionListParsers.cpp

```diff
@@ -1667,6 +1667,11 @@ class TrimLayer : public Layer
                 if (!mergeElement())
                     return false;
 
+                /// Skip replaceRegex if trying to TRIM an empty string to avoid exception
+                if (char_override && elements[0]->getID() == "Literal_''") {
```

Comment:
String comparison is not good (Perl-style), replace with checking the type and checking the literal value exactly.